core/src/main/java/org/apache/struts2/dispatcher/Dispatcher.java

```diff
@@ -88,7 +88,7 @@
      */
     public static final String REQUEST_POST_METHOD = "POST";
 
-    public static final String MULTIPART_FORM_DATA_REGEX = "^multipart\\/form-data(; boundary=[\\-a-zA-Z0-9]{1,70})?";
+    public static final String MULTIPART_FORM_DATA_REGEX = "^multipart/form-data(; boundary=[0-9a-zA-Z'()+_,\\-./:=?]{1,70})?";
```

Comment:
you are correct about the `-` needing to be escaped , but that's the only one i escaped. (they're in the order they're mentioned in RFC1341 (so i moved the `-` backwards in the pattern)